extide/gradle/src/org/netbeans/modules/gradle/ActionProviderImpl.java

```diff
@@ -123,15 +124,42 @@ public ActionProviderImpl(Project project) {
         for (GradleActionsProvider provider : providers) {
             actions.addAll(provider.getSupportedActions());
         }
+        // add a fixed 'prime build' action
+        actions.add(ActionProvider.COMMAND_PRIME);
         return actions.toArray(new String[actions.size()]);
     }
-
+    
     @Override
     public void invokeAction(String command, Lookup context) throws IllegalArgumentException {
         if (COMMAND_DELETE.equals(command)) {
             DefaultProjectOperations.performDefaultDeleteOperation(project);
             return;
         }
+        if (ActionProvider.COMMAND_PRIME.equals(command)) {
+            // delegate to prooblem provider we know exists & is registered
+            NbGradleProjectImpl prjImpl = project.getLookup().lookup(NbGradleProjectImpl.class);
+            ActionProgress prg = ActionProgress.start(context);
+            LOG.log(Level.FINER, "Priming build starting for {0}", project);
+            if (prjImpl.isProjectPrimingRequired()) {
+                prjImpl.primeProject().
+                        thenAccept(gp -> {
+                            LOG.log(Level.FINER, "Priming build of {0} finished with status {1}, ", new Object[] { project, prjImpl.isProjectPrimingRequired() });
+                            prg.finished(prjImpl.isProjectPrimingRequired());
+                        }).
+                        exceptionally((e) -> { 
+                            LOG.log(Level.FINER, e, () -> String.format("Priming build errored: %s", project));
```

Comment:
The previous `LOG.log` message is using curly braces pattern. This one is using lambda callback. I'd welcome consistency and I prefer to use the curly braces in both places.